From 28e04f33edc99a439ebe3db9f76a1d1d65439c4e Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Thu, 7 Aug 2025 11:19:20 +0200
Subject: [PATCH] Fix lvutil.extractUuid

It's not supposed to return SR UUID but is used for
parsing path like this
"VG_XenStorage--a6f308d1--764f--f50c--dcda--0d9c0be2d710-VHD--cd1d86c1--f379--400c--b1f4--5a73f74119d3"
instead. Replace its usage to obtain SR UUID in qcow2util by the
same log instead.

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/lvmcowutil.py |  3 +--
 drivers/qcow2util.py  | 10 ++++++----
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/drivers/lvmcowutil.py b/drivers/lvmcowutil.py
index 10d1607f..b4d24cf9 100755
--- a/drivers/lvmcowutil.py
+++ b/drivers/lvmcowutil.py
@@ -229,8 +229,7 @@ class LvmCowUtil(object):
         if uuid.startswith(VG_PREFIX):
             # we are dealing with realpath
             uuid = uuid.replace("--", "-")
-            uuid = uuid.replace(VG_PREFIX, "")
-            return uuid
+
         for prefix in LV_PREFIX.values():
             if uuid.find(prefix) != -1:
                 uuid = uuid.split(prefix)[-1]
diff --git a/drivers/qcow2util.py b/drivers/qcow2util.py
index 5622bb43..f344c87e 100644
--- a/drivers/qcow2util.py
+++ b/drivers/qcow2util.py
@@ -14,7 +14,7 @@ import xs_errors
 from blktap2 import TapCtl
 from cowutil import CowUtil, CowImageInfo
 from lvmcache import LVMCache
-from LVMSR import NS_PREFIX_LVM
+from constants import NS_PREFIX_LVM, VG_PREFIX
 
 MAX_QCOW_CHAIN_LENGTH: Final = 30
 
@@ -495,12 +495,14 @@ class QCowUtil(CowUtil):
             return None
 
         vdiUuid = extractUuidFunction(lvPath)
-        srUuid = extractUuidFunction(vgName)
-        lvcache.activate(NS_PREFIX_LVM + srUuid, vdiUuid, lvName, False)
+        srUuid = vgName.replace(VG_PREFIX, "")
+
+        ns = NS_PREFIX_LVM + srUuid
+        lvcache.activate(ns, vdiUuid, lvName, False)
         try:
             cowinfo = self.getInfo(lvPath, extractUuidFunction)
         finally:
-            lvcache.deactivate(NS_PREFIX_LVM + srUuid, vdiUuid, lvName, False)
+            lvcache.deactivate(ns, vdiUuid, lvName, False)
         return cowinfo
 
     @override
